packages/ses/src/error/console.js

```diff
@@ -362,12 +394,51 @@ const makeCausalConsole = (baseConsole, loggedErrorHandler) => {
   return /** @type {VirtualConsole} */ (freeze(causalConsole));
 };
 freeze(makeCausalConsole);
-export { makeCausalConsole };
 
-// /////////////////////////////////////////////////////////////////////////////
+/**
+ * @typedef {(...args: unknown[]) => void} Logger
+ */
+
+/**
+ * @param {LoggedErrorHandler} loggedErrorHandler
+ */
+export const defineCausalConsoleFromLogger = loggedErrorHandler => {
+  /**
+   * Implement the `VirtualConsole` API badly by turning all calls into
+   * calls on `tlogger`. We need to do this to have `console` logging
+   * turn into calls to Ava's `t.log`, so these console log messages
+   * are output in the right place.
+   *
+   * @param {Logger} tlogger
+   * @returns {VirtualConsole}
+   */
+  const makeCausalConsoleFromLogger = tlogger => {
+    const baseConsole = {};
+    for (const [name1, _] of consoleLevelMethods) {
+      baseConsole[name1] = tlogger;
+    }
+    for (const [name2, _] of [
+      ['group', 'log'],
+      ['groupCollapsed', 'log'],
+      ...consoleOtherMethods,
+    ]) {
+      baseConsole[name2] = (...args) => tlogger(name2, ...args);
+    }
```

Comment:
Suggestion: add indentation rather than "groupEnd" entries.
```suggestion
    const indents = [];
    const logWithIndent = (...args) => {
      if (indents.length > 0) {
        args = [...indents, ...args];
      }
      return tlogger(...args);
    };
    const makeNamed = (name, fn) =>
      ({ [name]: (...args) => fn(...args) })[name];
  
    const baseConsole = fromEntries([
      ...arrayMap(consoleLevelMethods, ([name]) => [
        name,
        makeNamed(name, logWithIndent),
      ]),
      ...arrayMap(consoleOtherMethods, ([name]) => [
        name,
        makeNamed(name, (...args) => logWithIndent(name, ...args)),
      ]),
    ]);
    // https://console.spec.whatwg.org/#grouping
    for (const name of ['group', 'groupCollapsed', 'groupEnd']) {
      if (!baseConsole[name]) continue;
      baseConsole[name] = makeNamed(name, (...args) => {
        if (name === 'groupEnd') {
          indents.pop();
          return;
        }
        // Prefix the logged data with "group" or "groupCollapsed".
        logWithIndent(name, ...args);
        // A single space is enough;
        // the host console will separate them with additional spaces.
        indents.push(' ');
      });
    }

```

Reply:
Oh geez there was a fly in the ointment on this one. A stack trace is multiple lines, i.e., with internal newlines. The logic above left all but the first line unindented.

Rather than find a way to fix the well, I just kludged until I got something acceptable, with a TODO explaining the problems to be fixed. I was tempted to just skip this suggestion to avoid the kludge. But the indented form is so much more pleasant to look at that it was worth it, thanks!

Kludgily done.